tests/Integration/PrestaShopBundle/TranslationIntegrationTest.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Tests\Integration\PrestaShopBundle;
+
+use PrestaShopBundle\Translation\PrestaShopTranslatorTrait;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+
+class TranslationIntegrationTest extends KernelTestCase
+{
+    use PrestaShopTranslatorTrait;
+
+    private $translator;
+
+    public function setUp(): void
+    {
+        self::bootKernel();
+        $this->translator = self::$kernel->getContainer()->get('translator');
+    }
+
+    public function testTranslator()
+    {
+        self::assertEquals('Succesful deletion', $this->translator->trans('Succesful deletion', ['legacy' => 'htmlspecialchars'], 'Admin.Notifications.Success', null));
+        self::assertEquals('The root category of the shop myshop', $this->translator->trans('The root category of the shop %shop%', ['%shop%' => 'myshop'], 'AdminCatalogNotification', null));
+        self::assertEquals('A new order has been placed on your shop', $this->translator->trans('A new order has been placed on your shop', [], 'Admin.Navigation.Header', null));
+        self::assertEquals('Image is too large (10 kB). Maximum allowed: 5 kB', $this->translator->trans('Image is too large (%1$d kB). Maximum allowed: %2$d kB', [10, 5], 'AdminNavigationSearch', null));
+        self::assertEquals('10 results match your query "stringTest".', $this->translator->trans('%d results match your query "%s".', [10, 'stringTest'], 'Admin.Navigation.Header', null));
+        self::assertEquals(
+            'This value is too short. It should have 3 characters or more.',
+            $this->translator->transChoice(
```

Comment:
This is ignoring the initial problem of `trans` ^^

This test class can indeed test both methods `trans` and `transChoice` but the main problem was that `trans` didn't handle plural messages correctly because of PrestaShop modifications in the trait, so you are still missing one use case here

Reply:
I'll try and fetch your branch in the afternoon to understand what was failing when you showed me